android/expoview/src/main/java/host/exp/exponent/kernel/services/ErrorRecoveryManager.java

```diff
@@ -14,6 +14,7 @@
 import host.exp.exponent.kernel.ExperienceId;
 import host.exp.exponent.storage.ExponentSharedPreferences;
 
+// todo: Remove this once SDK34 is phased out
```

Comment:
There's a whole bunch of code in Kernel depending on `ErrorRecoveryManager`, like `ReactNativeActivity#shouldShowErrorScreen` (called by `BaseExperienceActivity#consumeErrorQueue` (called by`BaseExperienceActivity.addError`…))

and `Updates` depending on `app.json#updates.checkAutomatically === 'ON_ERROR_RECOVERY'`

https://github.com/expo/expo/blob/cfa00db583bc9ffda1765a4a480c00c67dfbabab/android/expoview/src/main/java/host/exp/exponent/AppLoader.java#L119

Maybe we should annotate that code too with a clear strategy on how to deprecate it? Currently I don't see any support kept for `ON_ERROR_RECOVERY` setting for Updates. How do you handle it in `expo-ota` PR, @mczernek?

Reply:
I've tried to read and understood `ErrorRecoveryManager` code.
Here is what this class does:
- Android
  - owns recovery props
  - has `shouldReloadOnError` methods, which indicate if experience should reload after the crash (for me it's a little bit strange because this method depends on the time between app loading and crashing).
- iOS
  - associates arbitrary developer info with experience id (we don't use this)
  - knows if auto-reload after an error is possible 
  - tracks state of experience (`is recovering, has errors, last loaded`)
  - knows if error screen should be shown

Imho, we should leave most of this class. However, maybe renaming will be good options. 
This class has more common with error handling than with error recovering.

Reply:
So to sum up, `ErrorRecoveryManager`'s responsibilities are:
- **recovery props memory** — moved to native module
- **knows whether to show error screen based on timing of crashes/successful loads** — this can stay even longer than when SDK34 is phased out, we may want to leave this feature. Moreover, it is independent from the recovery props memory responsibility
- **informs other pieces of the infrastructure (`Updates`) that the experience is erroring** — this, fortunately, is also independent from the recovery props memory responsibility.

`ErrorRecoveryManager` doesn't need to be removed, it can stay as a part of ExpoKit. We just removed the memory responbility. 🙂